main.go

```diff
@@ -138,23 +144,47 @@ func main() {
 		}()
 	}
 
+	var logStream io.Writer
+	if *logFile != "" {
+		handle, err := os.OpenFile(*logFile, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0o644)
+		if err != nil {
+			setupLog.Error(fmt.Errorf("unable to open log file %s: %w", *logFile, err), "error initializing logging")
+		}
+		defer handle.Close()
+		logStream = handle
```

Comment:
if there is an err above, the `handle` will be `nil` right? won't this cause problems when attempting to close?

Reply:
It would return ErrInvalid, which would be silent:

https://cs.opensource.google/go/go/+/refs/tags/go1.19.4:src/os/file_posix.go;l=23

Because an error implies nothing was opened, presumably there is nothing to close.

However, a failure to open a logfile should probably be a fatal error, so updating to exit immediately on error.